pkg/k8shandler/kibana/reconciler.go

```diff
@@ -74,6 +107,10 @@ func ReconcileKibana(requestCluster *kibana.Kibana, requestClient client.Client,
 		return err
 	}
 
+	if err := clusterKibanaRequest.reIndexKibana(); err != nil {
```

Comment:
Might I suggest an alternate pattern here.  This is interlaced with reconcilation and basically is a one off process.  Consider introducing something like a "migration" package for migrating data where we can register specific migrations  and the logic is encapsulated entirely.  So maybe we have a "kibana5To6Migration"  that we can drop wholesale in future without needing to update the reconcile logic

Reply:
This is a good point. I put the migrations to a separate package. PTAL again.